### PR TITLE
add python solution 991

### PR DESCRIPTION
**Pull request by jjliewie**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-991.mdx [hunk 1]
```diff
@@ -67,5 +67,51 @@ public class Loan {
...
+		g += y * calc
+		cur -= calc
+
+	return g >= n
+
+low, high = 1, n + 1
+
+while low + 1 < high:
+
+	middle = low + high >> 1
```
Comment (ryanchou-dev) [inline solutions/usaco-991.mdx:106]: just keep it as `// 2`, easier to read.

File: solutions/usaco-991.mdx [hunk 2]
```diff
@@ -67,5 +67,51 @@ public class Loan {
...
+	g, cur = 0, k
+    
+	while g < n and cur:
+		y = math.floor((n - g) / mid)
+
+		if y <= m:
+			g += m * cur
+			break
+
+		calc = min(math.floor((n - mid * y - g) / y + 1), cur)
```
Comment (ryanchou-dev) [inline solutions/usaco-991.mdx:95]: floor division is default right

File: solutions/usaco-991.mdx [hunk 3]
```diff
@@ -67,5 +67,51 @@ public class Loan {
...
+<PySection>
+
+```python
+import sys
+import math
+
+sys.stdin = open("loan.in", "r")
+sys.stdout = open("loan.out", "w")
+
+input = sys.stdin.readline
```
Comment (ryanchou-dev) [inline solutions/usaco-991.mdx:80]: i commented about this on your other pr...why?

File: solutions/usaco-991.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-991.mdx:95]: its not

File: solutions/usaco-991.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-991.mdx:80]: ill change it

File: solutions/usaco-991.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-991.mdx:95]: `//`?

File: solutions/usaco-991.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-991.mdx:106]: seconded

File: solutions/usaco-991.mdx (hunk 3 above)
Comment (SansPapyrus683) [reply solutions/usaco-991.mdx:80]: `with`

File: solutions/usaco-991.mdx [hunk 4]
```diff
@@ -67,5 +67,51 @@ public class Loan {
...
+
+n, k, m = map(int, input().split())
+
+def check_which(mid):
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:84]: take out if possible

File: solutions/usaco-991.mdx [hunk 5]
```diff
@@ -67,5 +67,51 @@ public class Loan {
...
+
+	g, cur = 0, k
+    
+	while g < n and cur:
+		y = math.floor((n - g) / mid)
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:89]: you could also just do `(n - g) // mid`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-991.mdx (hunk 2 above)
Comment (jjliewie) [reply solutions/usaco-991.mdx:95]: oh mybad

File: solutions/usaco-991.mdx (hunk 3 above)
Comment (jjliewie) [reply solutions/usaco-991.mdx]: yep

File: solutions/usaco-991.mdx (hunk 5 above)
Comment (jjliewie) [reply solutions/usaco-991.mdx]: yep

File: solutions/usaco-991.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-991.mdx:99]: bump

File: solutions/usaco-991.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-991.mdx:77]: bump

File: solutions/usaco-991.mdx [hunk 6]
```diff
@@ -67,5 +67,45 @@ public class Loan {
...
+	g, cur = 0, k
+    
+	while g < n and cur:
+		y = (n - g) // mid
+
+		if y <= m:
+			g += m * cur
+			break
+
+		calc = min((n - mid * y - g) // y + 1, cur)
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:88]: comment, wth does this do

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
bump ^^

**Comment by ryanchou-dev** [comment]
stop

**Comment by ryanchou-dev** [comment]
omg stop

File: solutions/usaco-991.mdx (hunk 6 above)
Comment (jjliewie) [reply solutions/usaco-991.mdx:84]: calculate

Comment (Aadit-Ambadkar) [reply solutions/usaco-991.mdx:85]: calculate what? expand on it

**Comment by jjliewie** [comment]
> Does the code even work?

i just committed all of your suggestions.. I'll check rn. It worked before.

File: solutions/usaco-991.mdx [hunk 7]
```diff
@@ -67,5 +67,39 @@ public class Loan {
...
+low, high = 1, n + 1
+
+while low + 1 < high:
+	middle = low + high >> 1
+	if check_which(middle):
+		low = middle
+	else:
+		high = middle
+
+print(low, file=open('loan.out', 'w'))
```
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-991.mdx]: no

File: solutions/usaco-991.mdx [hunk 8]
```diff
@@ -92,7 +92,7 @@ low = 1
 high = n + 1
 
 while low + 1 < high:
-	middle = low + high >> 1
+	middle = (low + high) // 1
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:95]: wait what
you're dividing by 1?

Comment (jjliewie) [reply solutions/usaco-991.mdx:95]: mybad i just committed aadit's suggestion.. i assumed it was correct

Comment (jjliewie) [reply solutions/usaco-991.mdx]: changed it back

Comment (SansPapyrus683) [reply solutions/usaco-991.mdx]: or just
```suggestion
	middle = (low + high) // 2
```

**Comment by SansPapyrus683** [comment]
can you at least
address the changes

**Comment by jjliewie** [comment]
> can you at least address the changes

yeah I'm sorry i'll do it tonight

File: solutions/usaco-991.mdx [hunk 9]
```diff
@@ -67,5 +67,40 @@ public class Loan {
...
   
 </JavaSection>
+
+<PySection>
+
+```python
+with open('loan.in') as read:
+	n, k, m = map(int, read.readline().split())
+
+def check(mid):
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:77]: what does this function do?

File: solutions/usaco-991.mdx [hunk 10]
```diff
@@ -67,5 +67,40 @@ public class Loan {
...
+	g, cur = 0, k
+	while g < n and cur:
+		y = (n - g) // mid
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:80]: what the frick is this doing?

File: solutions/usaco-991.mdx [hunk 11]
```diff
@@ -67,5 +67,40 @@ public class Loan {
...
+		if y <= m:
+			g += m * cur
```
Comment (SansPapyrus683) [inline solutions/usaco-991.mdx:82]: and this?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-991.mdx (hunk 9 above)
Comment (SansPapyrus683) [reply solutions/usaco-991.mdx:78]: not really addressed
check if what's possible?
also type annotations

Comment (jjliewie) [reply solutions/usaco-991.mdx:78]: check if loan is possible

Comment (SansPapyrus683) [reply solutions/usaco-991.mdx:78]: type annotations

**Pull request closed without merging**